pandas-stubs/core/generic.pyi

```diff
@@ -305,10 +305,22 @@ class NDFrame(indexing.IndexingMixin):
     @overload
     def drop(
```

Comment:
I think the signatures for `DataFrame.drop()` and `Series.drop()` are all the same, so that means we can delete the definitions of `drop()` from `frame.pyi` and `series.pyi` and then put all of your changes in `generic.pyi` , and return `Self` in there.

I believe that the use of `HashableT1`, `HashableT2`, etc . in `series.pyi` for `drop()` were unnecessary, so let's use this opportunity to consolidate.

Reply:
Currently generic has `Hashable | Sequence[Hashable]` while frame has `Hashable | Sequence[Hashable] | Index[Any]` and series has `Hashable | list[HashableT2] | Index`.

So yeah, i think we can just use `Hashable | Sequence[Hashable] | Index[Any]` in generic and remove the overwrites in frame and series.

Reply:
added the `Index[Any]` to the signature in generic and removed the ones in series and frame.